libzeth/core/utils.cpp

```diff
@@ -40,41 +46,64 @@ static char nibble_hex(const uint8_t nibble)
     return '0' + nibble;
 }
 
-std::string hex_to_bytes(const std::string &s)
+// Return a pointer to the beginning of the actual hex characters (removing any
+// `0x` prefix), and ensure that the length is as expected.
+static const char *find_hex_start(const std::string &hex, const size_t bytes)
```

Comment:
I prefer to avoid `sanitize` here since that implies we are acting on the string itself and that we are doing some kind of thorough check (e.g. that all characters are valid hex).  I've renamed to `find_hex_string_of_length`.  Let me know if you prefer something else - I may have misunderstood the original objection.